src/js/components/Menu/Menu.js

```diff
@@ -368,16 +368,22 @@ const Menu = forwardRef((props, ref) => {
         // eslint-disable-next-line react/no-array-index-key
         key={groupIndex}
         {...theme.menu.group?.container}
-        border={
-          groupIndex > 0
-            ? {
-                side: 'top',
-                color: theme.menu.group?.separator?.color,
-                size: theme.menu.group?.separator?.size,
-              }
-            : undefined
-        }
       >
+        <Box
+          margin={{
```

Comment:
I've seen issues with margin crop up in the past where on areas of restricted width, the margin isn't respected. I'm wondering if we should implement as padding potentially:
```
<Box pad={{ horizontal: 'small' }}><Box border="top" /></Box>

Reply:
Good call, I updated this in the latest commit